### PR TITLE
CI: Add cache for Micromamba installation

### LINKED ISSUE #56: CI: Transient failures during Micromamba installation
[Example CI error](https://github.com/nextstrain/conda-base/actions/runs/7560882852/job/20587875591#step:8:13):

```
/Users/runner/work/_temp/a940fd79-97b7-42b8-9ac9-4d8ecc52367a.sh: line 1: micromamba: command not found
Error: Process completed with exit code 127.
```

This happens because the "Install Micromamba" step shows:

```
curl: (6) Could not resolve host: micro.mamba.pm
```

### Possible solutions

- Cache the Micromamba installation

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

This endpoint sometimes fails to resolve, so caching as a fallback should help.

## Related issue(s)

- Resolves #56 

## Checklist

- [ ] Cache hit and updated during CI re-run
- [ ] Checks pass

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -38,10 +38,21 @@ jobs:
     name: build and test (os=${{ matrix.os }})
     runs-on: ${{ matrix.os }}
     steps:
+      - name: Cache ~/.local/bin
+        id: cache-local-bin
+        uses: actions/cache@v4
+        with:
+          path: ~/.local/bin
+          # Cache should be valid for a maximum of 7 days.
```
Comment (corneliusroemer) [inline .github/workflows/ci.yaml:46]:
```suggestion
          # Cache should remain valid for 7 days from last usage
```

Comment (corneliusroemer) [reply .github/workflows/ci.yaml:46]: I read the docs differently: that it'll stay valid as long as you access it every 7 days

We should use a cache key so we get new versions from mamba.

**Review by corneliusroemer — DISMISSED** [review]
Thanks for figuring out a way to reduce failure rate.

This way we might never get updates from mamba.

Should we maybe use the cache only if the install fails? 

Or run the cache first, then install second even if the cache hit so we pull changes?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -38,10 +38,21 @@ jobs:
...
+          # <https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy>
+          key: ${{ runner.os }}-local-bin
+
       - name: Install Micromamba
-        run: |
-          curl -fsSL --proto 'https=' https://micro.mamba.pm/install.sh | bash
-          echo ~/.local/bin >> "$GITHUB_PATH"
+        if: steps.cache-local-bin.outputs.cache-hit != 'true'
```
Comment (corneliusroemer) [inline .github/workflows/ci.yaml:51]:
```suggestion
        if: steps.cache-local-bin.outputs.cache-hit != 'true'
```

If we get rid of this we should be able to use the cache as fallback and keep getting new micromamba versions

Comment (victorlin) [reply .github/workflows/ci.yaml:51]: Great suggestion!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (victorlin) [reply .github/workflows/ci.yaml:46]: Oh you're right, I'll update.

Comment (victorlin) [reply .github/workflows/ci.yaml]: Updated in 15d77232f0cb6e8f64e7d710763daf010b79ebd6.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (victorlin) [reply .github/workflows/ci.yaml]: Done in 15d77232f0cb6e8f64e7d710763daf010b79ebd6.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -38,11 +38,26 @@ jobs:
...
       - name: Install Micromamba
         run: |
           curl -fsSL --proto 'https=' https://micro.mamba.pm/install.sh | bash
           echo ~/.local/bin >> "$GITHUB_PATH"
 
+      - name: Save Micromamba to cache
+        uses: actions/cache/save@v4
+        with:
+          path: ~/.local/bin/micromamba
+          key: ${{ runner.os }}-micromamba
```
Comment (victorlin) [inline .github/workflows/ci.yaml:59]: This needs some more work - [it didn't save](https://github.com/nextstrain/conda-base/actions/runs/7563832066/job/20596953665?pr=57).

> Failed to save: Unable to reserve cache with key Linux-micromamba, another job may be creating this cache. More details: Cache already exists. Scope: refs/pull/57/merge, Key: Linux-micromamba, Version: 53670150c936a15bf347fc758f8d4ffb00bfaed7e1251039dfd6945875b46746
> Warning: Cache save failed.

**Comment by tsibley** [comment]
Caching this way seems error prone to me, so I'd personally avoid it.

If resolving `micro.mamba.pm` is a repeat issue (but other resolution works fine…?), then we could try fetching https://raw.githubusercontent.com/mamba-org/micromamba-releases/main/install.sh instead of https://micro.mamba.pm/install.sh.

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (corneliusroemer) [reply .github/workflows/ci.yaml:59]: oh no, that's a shame :(

**Comment by victorlin** [comment]
Good suggestion @tsibley. Closing in favor of #58

**Pull request closed without merging**